kitchensink/src/app/common/PageElements/FoundationElements/ColorRow/ColorRow.jsx

```diff
@@ -73,13 +74,17 @@ const ColorRow = (props) => (
                         darkFont={color.darkFont}
                         bordered={color.bordered}
                         stackPosition={color.stack}
-                        infoBox={(
-                          props.showInfo && (props.showInfo === 'fullInfo')
-                            ? { infoTitle: color.colorName, infoText: color.name, infoLabel: color.infoLabel }
-                            : (props.showInfo === 'onlyTitles')
-                              ? color.colorName ? { infoTitle: color.colorName } : {}
-                              : {}
-                        )}
+                        infoBox={(function () {
```

Comment:
[IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) for the best !
Yare fully right, nested ternary operation are not recommended by [airbnb code style](https://github.com/airbnb/javascript#comparison--nested-ternaries).

But I actually comes to something a bit different :

```
if(props.showInfo && props.showInfo === 'fullInfo') {
 return { infoTitle: color.colorName, infoText: color.name, infoLabel: color.infoLabel };
}
else{
  if(props.showInfo === 'onlyTitles'){
      if(color.colorName){
          return  { infoTitle: color.colorName };
      }
      else{
          return {};
      }
  }
  else{
      return {}:
  }
}
```

Reply:
I was of the opinion that we should try to use less 'else' statements

Reply:
> I was of the opinion that we should try to use less 'else' statements

Yes you are right. You can remove this else if you d like to. However, your function is quite different than mine. So it shows that the `else` statement still help for reading at some point.

```
  if (props.showInfo) {
       if (props.showInfo === 'fullInfo') {
          return { infoTitle: color.colorName, infoText: color.name, infoLabel: color.infoLabel };
       }
       if (props.showInfo === 'onlyTitles') {
         if (color.colorName) {
           return { infoTitle: color.colorName };
         }
      }
   }
    return {};
  }
```

Reply:
So we leave it

> > I was of the opinion that we should try to use less 'else' statements
> 
> @sanmiawi Yes you are right. You can remove this else if you d like to. Probably our functions are same right ?
> 
> ```
> (function() {
>   if (props.showInfo) {
>        if (props.showInfo === 'fullInfo') {
>           return { infoTitle: color.colorName, infoText: color.name, infoLabel: color.infoLabel };
>        }
>        if (props.showInfo === 'onlyTitles') {
>          if (color.colorName) {
>            return { infoTitle: color.colorName };
>          }
>        }
>   }
>   return {};
> })
> ```

So we leave it as it is?

Reply:
@sanmiawi I think the two functions do **not** do the same think.
Let's do a **simulations**. 

Let assume : `props.showInfo` is set to `'Hello sanmi'`.

With **my** function the returned value is `{}`.
In **your** function the return value is `undefined`.